### PR TITLE
Countries reports follow feature

### PR DESCRIPTION
**Pull request by MDikkii**

### CODE REVIEW

File: app/src/main/java/com/mobilecoronatracker/ui/countriesreports/CountriesListAdapter.kt [hunk 1]
```diff
@@ -6,13 +6,17 @@ import androidx.recyclerview.widget.RecyclerView
 import com.mobilecoronatracker.databinding.ItemCountryReportBinding
 import com.mobilecoronatracker.model.CountryReportModelable
 
-class CountriesListAdapter : RecyclerView.Adapter<CountriesListAdapter.ViewHolder>() {
+
+class CountriesListAdapter() :
```
Comment (m1awicki) [inline app/src/main/java/com/mobilecoronatracker/ui/countriesreports/CountriesListAdapter.kt]: do we need empty ctor here?

File: app/src/main/java/com/mobilecoronatracker/model/CountryReportModelable.kt [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
+    val country: String
+    val cases: Int
+    val todayCases: Int
+    val deaths: Int
+    val todayDeaths: Int
+    val recovered: Int
+    val critical: Int
+    val active: Int
+    var followed: Boolean
+    val hasMoreData: Boolean
```
Comment (m1awicki) [inline app/src/main/java/com/mobilecoronatracker/model/CountryReportModelable.kt]: We don't need that... yet ;)

File: app/src/main/res/layout/item_country_report.xml [hunk 3]
```diff
@@ -25,25 +40,37 @@
...
+            app:layout_constraintStart_toStartOf="@+id/start_guideline"
             app:layout_constraintTop_toTopOf="parent"
             tools:text="China" />
 
+        <ImageView
+            android:id="@+id/bookmark_button"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:src="@drawable/ic_flag"
+            android:onClick="@{() -> item.followed ? listener.onCountryUnfollowed(item.country) : listener.onCountryFollowed(item.country) }"
```
Comment (m1awicki) [inline app/src/main/res/layout/item_country_report.xml:53]: This will crash if listener won't be set.

**Review by m1awicki — CHANGES_REQUESTED** [review]

File: app/src/main/res/layout/item_country_report.xml (hunk 3 above)
Comment (MDikkii) [reply app/src/main/res/layout/item_country_report.xml:53]: It won't. Data binding is smart enough! :)

**Review by m1awicki — APPROVED** [review]

File: app/src/main/java/com/mobilecoronatracker/model/CountryReportModelable.kt (hunk 2 above)
Comment (MDikkii) [reply app/src/main/java/com/mobilecoronatracker/model/CountryReportModelable.kt]: Removed whole hasMoreData feature

File: app/src/main/java/com/mobilecoronatracker/ui/countriesreports/CountriesListAdapter.kt (hunk 1 above)
Comment (MDikkii) [reply app/src/main/java/com/mobilecoronatracker/ui/countriesreports/CountriesListAdapter.kt]: No, changed

**Pull request merged**